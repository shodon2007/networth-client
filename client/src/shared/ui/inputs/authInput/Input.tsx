import { InputHTMLAttributes, FC, ReactNode } from "react";
import cls from './authInput.module.scss'

interface InputProps extends InputHTMLAttributes<HTMLInputElement>{
  children: string
}

const Input: FC<InputProps> = ({children, ...props}) => {
  return <input className={cls.Input} {...props} placeholder={children} />
}

export default Input