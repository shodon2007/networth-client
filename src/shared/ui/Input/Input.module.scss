.Input {
	border-radius: var(--medium-border-radius);
	padding: 0px 12px;
	width: 400px;
	height: 60px;
	color: var(--white-color);

	background: var(--bg-secondary-color);
	transition: 0.2s;
	&:hover {
		background: var(--bg-secondary-hover-color);
	}

	&:focus {
		background: var(--bg-secondary-focus-color);
	}

	&::placeholder {
		font-family: var(--font-main-Blin);
		font-weight: 600;
		font-size: var(--font-size-l-mobile);
		color: rgba(255, 255, 255, 0.8);
	}
}
