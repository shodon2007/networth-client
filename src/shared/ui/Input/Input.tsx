import {InputHTMLAttributes, FC} from "react";
import cls from "./Input.module.scss";
import classNames from "src/shared/lib/classNames";

export enum InputSize {
	big = "BIG",
	small = "SMALL",
	medium = "MEDIUM",
}

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
	classname?: string;
	fontSize?: InputSize;
	placeholderOutside?: boolean;
}

const Input: FC<InputProps> = ({
	classname = "",
	fontSize = InputSize.medium,
	placeholderOutside = false,
	...props
}) => {
	if (placeholderOutside) {
		return (
			<>
				<div>{props.name}</div>
				<input
					className={classNames(cls.Input, {}, [classname, cls[fontSize]])}
					{...props}
				/>
			</>
		);
	}

	return (
		<input
			className={classNames(cls.Input, {}, [classname, cls[fontSize]])}
			{...props}
		/>
	);
};

export default Input;
