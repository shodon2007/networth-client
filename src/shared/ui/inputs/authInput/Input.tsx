import { InputHTMLAttributes, FC } from "react";
import cls from './authInput.module.scss'
import classNames from "src/shared/lib/classNames";

interface InputProps extends InputHTMLAttributes<HTMLInputElement>{
  children: string,
  classname?: string,
}

const Input: FC<InputProps> = ({children, classname, ...props}) => {
  return <input  className={classNames(cls.Input, {}, [])} {...props} placeholder={children} />
}

export default Input