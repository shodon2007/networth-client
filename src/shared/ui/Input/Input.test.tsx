import {render} from "@testing-library/react";
import Input, {InputSize} from "./Input";

describe("Testing Input.tsx", () => {
	it("should pass snapshot", () => {
		const screen = render(<Input />);
		const input = screen.getByTestId("input-parent");
		expect(input).toBeInTheDocument();
		expect(input).toMatchSnapshot();
	});
	test("Testing placeholderOutline mode", () => {
		const screen = render(<Input placeholderOutside={true} />);
		const div = screen.getByTestId("input-placeholder");
		expect(div).toBeInTheDocument();
	});
	test("Testing no placeholderOutline mode", () => {
		const screen = render(<Input placeholderOutside={false} />);
		const div = screen.queryByTestId("input-placeholder");
		expect(div).toBeNull();
	});
	describe("Testing and snapshoting input sizes", () => {
		test("Testing small input", () => {
			const smallInput = render(<Input fontSize={InputSize.small} />);
			const smallInputDom = smallInput.getByTestId("input-parent");
			expect(smallInputDom).toBeInTheDocument();
			expect(smallInputDom).toMatchSnapshot();
		});
		test("Testing small input", () => {
			const mediumInput = render(<Input fontSize={InputSize.medium} />);
			const mediumInputDom = mediumInput.getByTestId("input-parent");
			expect(mediumInputDom).toBeInTheDocument();
			expect(mediumInputDom).toMatchSnapshot();
		});
		test("Testing small input", () => {
			const bigInput = render(<Input fontSize={InputSize.big} />);
			const bigInputDom = bigInput.getByTestId("input-parent");
			expect(bigInputDom).toBeInTheDocument();
			expect(bigInputDom).toMatchSnapshot();
		});
	});
});
