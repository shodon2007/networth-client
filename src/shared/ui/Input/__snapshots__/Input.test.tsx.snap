// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input.tsx Snapshoting input sizes 1`] = `
<div
  data-testid="input-parent"
>
  <input
    class="input small"
    data-testid="input"
  />
</div>
`;

exports[`Testing Input.tsx Testing and snapshoting input sizes Testing small input 1`] = `
<div
  data-testid="input-parent"
>
  <input
    class="input small"
    data-testid="input"
  />
</div>
`;

exports[`Testing Input.tsx Testing and snapshoting input sizes Testing small input 2`] = `
<div
  data-testid="input-parent"
>
  <input
    class="input medium"
    data-testid="input"
  />
</div>
`;

exports[`Testing Input.tsx Testing and snapshoting input sizes Testing small input 3`] = `
<div
  data-testid="input-parent"
>
  <input
    class="input BIG"
    data-testid="input"
  />
</div>
`;

exports[`Testing Input.tsx should pass snapshot 1`] = `
<div
  data-testid="input-parent"
>
  <input
    class="input medium"
    data-testid="input"
  />
</div>
`;
